src/helpers.ts

```diff
@@ -55,6 +55,37 @@ export class TurbopufferError extends Error {
   }
 }
 
+export function getUpdatedUrlPath(
+  baseUrl: string,
+  path: string,
+  query?: Record<string, string | undefined>,
+) {
```

Comment:
Only nit here is this could maybe be better named + have a doc comment explaining what it's doing

Reply:
agree about the naming, what's being updated here?